### PR TITLE
cni manifest upgrade downgrade test

### LINKED ISSUE #694: Automate CNI plugin upgrade testing
Before releases, we manually perform a test of the upgrade process for the CNI plugin:

1. create 3 c3.xlarge node cluster
1. scale pod replica from 0 to 130
1. verify kubectl get deployment
1. upgrade cni to new version
1. verify kubectl get deployment
1. scale replica from 130 to 0
1. verify kubectl get deployment
1. scale replica from 0 to 130 again and verify

We should automate this and gate all PRs on a successful upgrade test

- [ ] Convert above steps into a Bash script
- [x] Add job to CircleCI pipeline, gating for all PRs

### PR DESCRIPTION
**Pull request by cgchinmay**
**What type of PR is this?**
Upgrade/downgrade tests for user specified cni manifest file

feature

**Which issue does this PR fix**:
Fixes #694 

**What does this PR do / Why do we need it**:
Allows to test whether upgrade and downgrade of cni works correctly when installed using manifest file

**If an issue # is not available please add repro steps and logs from IPAMD/CNI showing the issue**:

**Testing done on this change**:

```
ginkgo -v -r -- --cluster-kubeconfig=/Users/cgadgil/.kube/config --cluster-name=cni-rc-test --aws-region=us-west-2 --aws-vpc-id=vpc-0faed7c6e93018b48 --ng-name-label-key=eks.amazonaws.com/nodegroup --ng-name-label-val=nodegroup --target-manifest-file=https://raw.githubusercontent.com/aws/amazon-vpc-cni-k8s/master/config/master/aws-k8s-cni.yaml --initial-manifest-file=https://raw.githubusercontent.com/aws/amazon-vpc-cni-k8s/v1.9.0/config/master/aws-k8s-cni.yaml

STEP: creating test namespace
STEP: getting the node with the node label key eks.amazonaws.com/nodegroup and value nodegroup
STEP: verifying more than 1 nodes are present for the test
STEP: getting the instance type from node label beta.kubernetes.io/instance-type
STEP: getting the network interface details from ec2
STEP: describing the VPC to get the VPC CIDRs
STEP: setting the environment variables on the ds to map[WARM_ENI_TARGET:0 WARM_IP_TARGET:3]
STEP: getting the aws-node daemon set in namesapce kube-system
STEP: updating the daemon set with new environment variable
test host networking when pods using IP from primary and secondary ENI are created 
  should have correct host networking setup when pods are running and cleaned up when pods are terminated
  /Users/cgadgil/Documents/amazon-vpc-cni-k8s/test/integration-new/cni-upgrade-downgrade/host_networking_test.go:30
STEP: Applying manifest: https://raw.githubusercontent.com/aws/amazon-vpc-cni-k8s/v1.9.0/config/master/aws-k8s-cni.yaml
clusterrolebinding.rbac.authorization.k8s.io/aws-node unchanged
clusterrole.rbac.authorization.k8s.io/aws-node unchanged
Warning: apiextensions.k8s.io/v1beta1 CustomResourceDefinition is deprecated in v1.16+, unavailable in v1.22+; use apiextensions.k8s.io/v1 CustomResourceDefinition
customresourcedefinition.apiextensions.k8s.io/eniconfigs.crd.k8s.amazonaws.com unchanged
daemonset.apps/aws-node configured
serviceaccount/aws-node unchanged
STEP: creating a deployment to launch pod using primary and secondary ENI IP
STEP: getting the list of pods using IP from primary and secondary ENI
...
...
...
STEP: update cni to target manifest
STEP: Applying manifest: https://raw.githubusercontent.com/aws/amazon-vpc-cni-k8s/master/config/master/aws-k8s-cni.yaml
serviceaccount/aws-node configured
customresourcedefinition.apiextensions.k8s.io/eniconfigs.crd.k8s.amazonaws.com configured
clusterrole.rbac.authorization.k8s.io/aws-node configured
clusterrolebinding.rbac.authorization.k8s.io/aws-node configured
daemonset.apps/aws-node configured
STEP: deleting the deployment to test teardown
STEP: waiting to allow CNI to tear down networking for terminated pods
STEP: validating host networking is teared down correctly
STEP: creating pod to test host networking setup
...
...
2022/02/15 03:32:40 found no rules for the pod's IP 192.168.2.103
2022/02/15 03:32:40 no leaked resource found for the pod cni-automation/deployment-test-bd765bfc5-pcnft

STEP: deleting the host networking setup pod

• [SLOW TEST:126.389 seconds]
test host networking
/Users/cgadgil/Documents/amazon-vpc-cni-k8s/test/integration-new/cni-upgrade-downgrade/host_networking_test.go:20
  when pods using IP from primary and secondary ENI are created
  /Users/cgadgil/Documents/amazon-vpc-cni-k8s/test/integration-new/cni-upgrade-downgrade/host_networking_test.go:29
    should have correct host networking setup when pods are running and cleaned up when pods are terminated
    /Users/cgadgil/Documents/amazon-vpc-cni-k8s/test/integration-new/cni-upgrade-downgrade/host_networking_test.go:30
------------------------------
STEP: deleting test namespace
STEP: update environment variables map[], remove map[WARM_ENI_TARGET:{} WARM_IP_TARGET:{}]
STEP: getting the aws-node daemon set in namesapce kube-system
STEP: updating the daemon set with new environment variable

Ran 1 of 1 Specs in 221.592 seconds
SUCCESS! -- 1 Passed | 0 Failed | 0 Pending | 0 Skipped
```

**Automation added to e2e**:

**Will this PR introduce any new dependencies?**:

errors package included

**Will this break upgrades or downgrades. Has updating a running cluster been tested?**:
No

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: scripts/upload-resources-to-github.sh [hunk 1]
```diff
@@ -1,249 +0,0 @@
-#!/bin/bash
-set -euo pipefail
```
Comment (jayanthvn) [inline scripts/upload-resources-to-github.sh:2]: Why is this file getting deleted?

File: test/integration-new/cni-upgrade-downgrade/host_networking_test.go [hunk 2]
```diff
@@ -0,0 +1,99 @@
+package cni_upgrade_downgrade
```
Comment (jayanthvn) [inline test/integration-new/cni-upgrade-downgrade/host_networking_test.go:1]: Can we include custom networking and PD mode test with upgrade/downgrade?

Comment (cgchinmay) [reply test/integration-new/cni-upgrade-downgrade/host_networking_test.go:1]: For Custom networking we create a new node group but for PD mode we should be able to run against existing cluster. Could you elaborate on the steps for how exactly we want the run the PD tests with upgrade/downgrade scenario.

Comment (cgchinmay) [reply test/integration-new/cni-upgrade-downgrade/host_networking_test.go]: Have added PD traffic test, this is the flow
apply initial manifest -> Set PD -> Run traffic test -> apply target manifest with PD enabled -> Run traffic test -> roll back to initial manifest -> reset PD.

Comment (jayanthvn) [reply test/integration-new/cni-upgrade-downgrade/host_networking_test.go]: Thank you!

**Review by jayanthvn — APPROVED** [review]
lgtm

**Pull request merged**